packages/canary-web-components/src/components/ic-tree-item/ic-tree-item.tsx

```diff
@@ -0,0 +1,355 @@
+      (isChild && !this.el.isParent)
+    ) {
+      if (isRouterItem) {
+        this.routerSlot.classList.add("ic-tree-item-single");
+      } else {
+        treeItemContent.classList.add("ic-tree-item-single");
+      }
+    }
+
+    while (parentElement) {
+      if (parentElement.tagName === this.treeItemTag) {
+        level++;
+        if (!this.el.isParent && isSiblingOfParent) {
+          treeItemContent.style.paddingLeft = `calc(var(--ic-space-xl) + ${
+            level * 16
+          }px)`;
+        } else if (!this.el.isParent) {
+          treeItemContent.style.paddingLeft = `calc(var(--ic-space-xs) + ${
+            level * 24
+          }px`;
+        } else {
+          treeItemContent.style.paddingLeft = `${level * 16}px`;
+        }
+      }
+      parentElement = parentElement.parentElement;
+    }
+  };
+
+  private truncateTreeItemLabel = (treeItem: HTMLIcTreeItemElement) => {
+    const typographyEl: HTMLIcTypographyElement =
+      treeItem.shadowRoot.querySelector(".tree-item-label");
+    const tooltip = typographyEl.closest("ic-tooltip");
+    const treeContent: HTMLElement =
+      treeItem.shadowRoot.querySelector(".tree-item-content");
+
+    if (typographyEl?.scrollHeight > treeContent?.clientHeight) {
+      typographyEl.classList.add("ic-text-overflow");
+
+      if (!tooltip) {
+        const tooltipEl = document.createElement("ic-tooltip");
+        tooltipEl.setAttribute("target", this.el.id);
+        tooltipEl.setAttribute("label", typographyEl.textContent);
+        tooltipEl.classList.add("ic-tooltip-overflow");
+        tooltipEl.setAttribute("placement", "right");
+        treeContent.appendChild(tooltipEl);
+        tooltipEl.appendChild(typographyEl);
+      }
+    }
+  };
+
+  private hasRouterSlot(): boolean {
+    this.routerSlot = this.el.querySelector('[slot="router-item"]');
+    if (this.routerSlot) {
+      this.routerSlot.ariaLabel = this.routerSlot.textContent;
+    }
+    return !!this.routerSlot;
+  }
+
+  private hostMutationCallback = (mutationList: MutationRecord[]): void => {
+    if (
+      mutationList.some(({ type, addedNodes, removedNodes }) =>
+        type === "childList"
+          ? checkSlotInChildMutations(addedNodes, removedNodes, "icon")
+          : false
+      )
+    ) {
+      forceUpdate(this);
+    }
+  };
+
+  render() {
+    const { appearance, disabled, label, selected, size, expanded } = this;
+
+    const Component = this.href ? "a" : "div";
+
+    const attrs = Component == "a" && {
+      href: this.href,
+      hrefLang: this.hreflang,
+      referrerPolicy: this.referrerpolicy,
+      rel: this.rel,
+      target: this.target,
+    };
+
+    return (
+      <Host
+        class={{
+          [`ic-tree-item-${appearance}`]: true,
+          "ic-tree-item-disabled": disabled,
+          "ic-tree-item-selected": selected,
+          [`ic-tree-item-${size}`]: size !== "default",
+        }}
+        aria-disabled={disabled ? "true" : "false"}
```

Comment:
When testing out the storybook in Voiceover, I'm able to move the virtual cursor to Disabled tree items (Ctrl+Opt+RightArrow) and then click them - [in the case of Links](https://mi6.github.io/ic-ui-kit/branches/758-tree-view/canary-react/?path=/story/react-components-tree-view--link) it activates the link. 
The disabled item doesn't say "dimmed" either, which disabled items normally do - maybe the aria-disabled isn't on the correct element?

Reply:
Had a look around and it turns out you can't disable a link, so I've updated it to stay as a div element if href is set and disabled is true, hopefully this is okay